spec/app/ics-027-interchain-accounts/README.md

```diff
@@ -175,40 +173,99 @@ An example of an active channel on the controller chain can look like this:
-| Relayer   | ChanOpenConfirm  | Host             |  (ics27-1, ics27-1.{interchain-account-address})        | 
+**INIT**
+
+Initiator: Controller
+
+Datagram: ChanOpenInit
+
+Chain Acted Upon: Controller
+
+Version: 
+```json
+{
+  "Version": "ics27-1",
+  "Encoding": ["supported encoding types", "..."],
+  "Tx_Types": ["supported transaction types", "..."],
+  "ControllerConnectionId": "self_connection_id",
+  "HostConnectionId": "counterparty_connection_id",
+  "Address": "",
+}
+```
+
+Comments: The address is left empty since this will be generated and relayed back by the host chain. The INIT step may include multiple supported encoding and transaction types.
 
-The channel `version` string passed into the `OnChanOpenTry` callback by a relayer will contain the account address of the interchain account that will be registered on the host chain. A relayer will then pass this version into the `OnChanOpenAck` step on the controller chain as the `counterpartyVersion`. The controller chain will then proceed to parse the interchain account address from the `counterpartyVersion` string and store this address in state with a map to the associated portID.  
+**TRY**
 
-#### Controlling flow
+Initiator: Relayer
+
+Datagram: ChanOpenTry
+
+Chain Acted Upon: Host
+
+Version: 
+```json
+{
+  "Version": "ics27-1",
+  "Encoding": "negotiated_encoding_type",
+  "Tx_Types": "negotiated_tx_type",
+  "ControllerConnectionId": "self_connection_id",
+  "HostConnectionId": "counterparty_connection_id",
+  "Address": "interchain_account_address",
+}
+```
+
+Comments: The ICS-27 application on the host chain is responsible for returning this version given the counterparty version set by the controller chain in INIT. The host chain must select a single encoding type and a single tx type that is supported by both the host chain and the controller chain (ie. included in counterparty version). If no such encoding or tx type exists, then the host chain must return an error and abort the handshake.
+The host chain must also generate the interchain account address and populate the address field in the version with the interchain account address string.
+
+**ACK**
+
+Initiator: Relayer
+
+Datagram: ChanOpenTry
+
+Chain Acted Upon: Host
+
+CounterpartyVersion: 
+```json
+{
+  "Version": "ics27-1",
+  "Encoding": "negotiated_encoding_type",
+  "Tx_Types": "negotiated_tx_type",
+  "ControllerConnectionId": "self_connection_id",
+  "HostConnectionId": "counterparty_connection_id",
+  "Address": "interchain_account_address",
+}
+```
+
+Comments: On the ChanOpenAck step, the ICS27 application on the controller chain must verify the version string chosen by the host chain on ChanOpenTry. The controller chain must verify that it can support the negotiated encoding and tx type selected by the host chain. If either is unsupported, then it must return an error and abort the handshake.
```

Comment:
```suggestion
Comments: On the `OnChanOpenAck` step, the ICS-27 application on the controller chain must verify the version string chosen by the host chain on `OnChanOpenTry`. The controller chain must verify that it can support the negotiated encoding and tx type selected by the host chain. If either is unsupported, then it must return an error and abort the handshake.
```